src/libkernel/sscp/amdgpu/math.cpp

```diff
@@ -111,11 +111,16 @@ HIPSYCL_SSCP_BUILTIN float __acpp_sscp_frexp_f32(float x, __acpp_int32 *y) {
   }
 }
 
-HIPSYCL_SSCP_BUILTIN double __acpp_sscp_frexp_f64(double x, __acpp_int64 *y) {
+HIPSYCL_SSCP_BUILTIN double __acpp_sscp_frexp_f64(double x, __acpp_int32 *y) {
   __acpp_int32 w;
-  double res = __ocml_frexp_f64(x, to_private(&w));
-  *y = static_cast<__acpp_int64>(w);
-  return res;
+  if(__ockl_is_private_addr(y))
+    return __ocml_frexp_f64(x, to_private(y));
+  else {
+    __acpp_int32 w;
+    float res = __ocml_frexp_f64(x, to_private(&w));
+    *y = w;
+    return res;
+  }
```

Comment:
Is there actually a performance benefit for this additional complexity? It looks to me like we're optimizing away one copy to `w` at the expense of more complex control flow, which might eat up the benefit.

Reply:
I copied this from the preexisting code above when I changed the second parameter type to match (and same for the other comment reference this). I did think of a way to recover performance which I'll discuss in the builtins comment.

```c++
HIPSYCL_SSCP_BUILTIN float __acpp_sscp_frexp_f32(float x, __acpp_int32 *y) {
  if(__ockl_is_private_addr(y))
    return __ocml_frexp_f32(x, to_private(y));
  else {
    __acpp_int32 w;
    float res = __ocml_frexp_f32(x, to_private(&w));
    *y = w;
    return res;
  }
}
```

Reply:
Oh, I see. I missed that this pattern already existed.